internal/cli/cli.go

```diff
@@ -0,0 +1,83 @@
+package cli
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/ioki-mobility/go-outline"
+	"github.com/spf13/cobra"
+	"net/http"
+)
+
+const (
+	flagNameBaseUrl = "baseUrl"
+	flagNameApiKey  = "apiKey"
+)
+
+func Run() error {
+	rootCmd := rootCmd()
+	collectionCmd := collectionCmd(rootCmd)
+	rootCmd.AddCommand(collectionCmd)
+
+	return rootCmd.Execute()
+}
+
+func rootCmd() *cobra.Command {
+	var rootCmd = &cobra.Command{Use: "outline"}
+
+	var apiKeyFlag string
+	var baseUrlFlag string
+	rootCmd.PersistentFlags().StringVar(&baseUrlFlag, flagNameBaseUrl, "", "The base url to the outline API instance (should include /api/ as suffix)")
+	rootCmd.PersistentFlags().StringVar(&apiKeyFlag, flagNameApiKey, "", "The outline apiKey")
+
+	return rootCmd
+}
+
+func collectionCmd(rootCmd *cobra.Command) *cobra.Command {
+	collectionCmd := &cobra.Command{
+		Use:   "collection",
+		Short: "Work with collections",
+		Long:  `If you have to work with collection in any case, use this command`,
+		Args:  cobra.MinimumNArgs(1),
+	}
+
+	printlnCommand := printlnCollectionCmd(rootCmd)
+	collectionCmd.AddCommand(printlnCommand)
+
+	return collectionCmd
+}
+
+func printlnCollectionCmd(rootCmd *cobra.Command) *cobra.Command {
+	return &cobra.Command{
+		Use:   "print",
+		Short: "Print the given collections (by ids)",
+		Long:  "Print the given collections (by ids)",
+		Args:  cobra.MinimumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			baseUrl, err := rootCmd.Flags().GetString(flagNameBaseUrl)
+			if err != nil {
+				panic(err)
+			}
+			apiKey, err := rootCmd.Flags().GetString(flagNameApiKey)
+			if err != nil {
+				panic(err)
+			}
+			outlineClient := outline.New(baseUrl, &http.Client{}, apiKey)
```

Comment:
naming the var as just `client` is enough IMO. We can see very easily its type and details very easily in a static language like go so why add extra noise :)